feat(api): allow moving labware with lids

sfoster1 opened the pull request:
Allow moving a labware (labware-bottom) with another labware (labware-top) on top of it, as long as labware-top is labware-bottom's lid by its lid id.

## Testing
- [x] try and move a tiprack that has a lid on it

```
metadata = {
    "protocolName": "Move tiprack with lid",
}

requirements = {
    "robotType": "OT-3",
    "apiLevel": "2.23",
}


def run(protocol):
    trash = protocol.load_trash_bin(location="A3")
    plate = protocol.load_labware("armadillo_96_wellplate_200ul_pcr_full_skirt", "C1")

    tiprack = protocol.load_labware(
        "opentrons_flex_96_tiprack_1000ul", "D1", lid="opentrons_flex_tiprack_lid"
    )
    adapter = protocol.load_adapter("opentrons_flex_96_tiprack_adapter", "D2")

    p1000 = protocol.load_instrument("flex_96channel_1000", "left", tip_racks=[tiprack])
    protocol.move_labware(tiprack, adapter, use_gripper=True)
    protocol.move_lid(tiprack, trash, use_gripper=True)
    p1000.pick_up_tip()
    p1000.aspirate(200, plate["A1"])
    p1000.dispense(200, plate["A1"])
    p1000.drop_tip()
```

Closes EXEC-1116

sfoster1 committed:
feat(api): allow moving labware with lids

Allow moving a labware (labware-bottom) with another
labware (labware-top) on top of it, as long as labware-top is
labware-bottom's lid by its lid id.

Closes EXEC-1116

sfoster1 committed:
format

smb2268 commented on shared-data/js/labware.ts:
```diff
@@ -301,6 +302,7 @@ const usascientific12Reservoir22MlV1 = usascientific12Reservoir22MlV1Uncasted as
 const usascientific96Wellplate24MlDeepV1 = usascientific96Wellplate24MlDeepV1Uncasted as LabwareDefinition2
 const evotipsFlex96TiprackAdapterV1 = evotipsFlex96TiprackAdapterV1Uncasted as LabwareDefinition2
 const evotipsOpentrons96LabwareV1 = evotipsOpentrons96LabwareV1Uncasted as LabwareDefinition2
+const opentronsFlexTiprackLidV1 = opentronsFlexTiprackLidV1Uncasted as LabwareDefinition2
```
Do we want these tiprack lids to be displayed in the labware tab of the app and in labware library alongside TC lids?

sfoster1 replied:
we do not! adding them to the blocklist

CaseyBatten reviewed (approved):
Engine implementation looks good, very cool labware.

sfoster1 committed:
fix(app): support lids in more places

- The tiprack lid def should be exported
- Various places that get lists of labware from analyses need to know to
  get them from loadLid as well as loadLabware

Hopefully no functional change aside from whitescreening less.

smb2268 commented on components/src/organisms/ProtocolTimelineScrubber/utils.ts:
```diff
@@ -152,7 +152,8 @@ export function getLabwareDefinitionsFromCommands(
     const isLoadingNewDef =
-      command.commandType === 'loadLabware' &&
+      (command.commandType === 'loadLabware' ||
+        command.commandType === 'loadLid') &&
```
we may also need to add `commandType === 'loadLidStack'` here and above

sfoster1 replied:
We definitely will need to add handling for that but I think the idea is that lid stack objects generally won't be shown in frontends. Can we do it in a follow up?

smb2268 replied:
We don't want the imaginary first stack item to be shown on the FE but we do want all of the lids that are loaded via a `loadLidStack` bulk command to be displayed. I'd be curious if we run into this same white screen issue if a lid is initially loaded through a `loadLidStack` and then moved onto another labware, which is then moved.
